src/devices/cpu/tlcs900/tmp94c241.h

```diff
@@ -0,0 +1,419 @@
+	void treg89_w(offs_t offset, uint16_t data);
+	void tregab_w(offs_t offset, uint16_t data);
+	uint8_t cap45_r(offs_t offset);
+	uint8_t cap67_r(offs_t offset);
+	uint8_t cap89_r(offs_t offset);
+	uint8_t capab_r(offs_t offset);
+	uint8_t t16run_r();
+	void t16run_w(uint8_t data);
+	uint8_t wdmod_r();
+	void wdmod_w(uint8_t data);
+	void wdcr_w(uint8_t data);
+	uint8_t sc0buf_r();
+	void sc0buf_w(uint8_t data);
+	uint8_t sc0cr_r();
+	void sc0cr_w(uint8_t data);
+	uint8_t sc0mod_r();
+	void sc0mod_w(uint8_t data);
+	uint8_t br0cr_r();
+	void br0cr_w(uint8_t data);
+	uint8_t sc1buf_r();
+	void sc1buf_w(uint8_t data);
+	uint8_t sc1cr_r();
+	void sc1cr_w(uint8_t data);
+	uint8_t sc1mod_r();
+	void sc1mod_w(uint8_t data);
+	uint8_t br1cr_r();
+	void br1cr_w(uint8_t data);
+	uint8_t ode_r();
+	void ode_w(uint8_t data);
+	uint8_t admod1_r();
+	void admod1_w(uint8_t data);
+	uint8_t admod2_r();
+	void admod2_w(uint8_t data);
+	uint8_t adreg_r(offs_t offset);
+	uint8_t inte_r(offs_t offset);
+	void inte_w(offs_t offset, uint8_t data);
+	uint8_t intnmwdt_r(offs_t offset);
+	void intnmwdt_w(offs_t offset, uint8_t data);
+	void iimc_w(uint8_t data);
+	void intclr_w(uint8_t data);
+	void dmav_w(offs_t offset, uint8_t data);
+	void b0cs_w(offs_t offset, uint8_t data);
+	void b1cs_w(offs_t offset, uint8_t data);
+	void b2cs_w(offs_t offset, uint8_t data);
+	void b3cs_w(offs_t offset, uint8_t data);
+	void b4cs_w(offs_t offset, uint8_t data);
+	void b5cs_w(offs_t offset, uint8_t data);
+	uint8_t msar0_r();
+	void msar0_w(offs_t offset, uint8_t data);
+	uint8_t mamr0_r();
+	void mamr0_w(offs_t offset, uint8_t data);
+	uint8_t msar1_r();
+	void msar1_w(offs_t offset, uint8_t data);
+	uint8_t mamr1_r();
+	void mamr1_w(offs_t offset, uint8_t data);
+	uint8_t msar2_r();
+	void msar2_w(offs_t offset, uint8_t data);
+	uint8_t mamr2_r();
+	void mamr2_w(offs_t offset, uint8_t data);
+	uint8_t msar3_r();
+	void msar3_w(offs_t offset, uint8_t data);
+	uint8_t mamr3_r();
+	void mamr3_w(offs_t offset, uint8_t data);
+	uint8_t msar4_r();
+	void msar4_w(offs_t offset, uint8_t data);
+	uint8_t mamr4_r();
+	void mamr4_w(offs_t offset, uint8_t data);
+	uint8_t msar5_r();
+	void msar5_w(offs_t offset, uint8_t data);
+	uint8_t mamr5_r();
+	void mamr5_w(offs_t offset, uint8_t data);
+	uint8_t drefcr1_r();
+	void drefcr1_w(uint8_t data);
+	uint8_t dmemcr1_r();
+	void dmemcr1_w(uint8_t data);
+	uint8_t drefcr3_r();
+	void drefcr3_w(uint8_t data);
+	uint8_t dmemcr3_r();
+	void dmemcr3_w(uint8_t data);
+	uint8_t dadrv_r();
+	void dadrv_w(uint8_t data);
+	void dareg_w(offs_t offset, uint8_t data);
+
+	void internal_mem(address_map &map);
+
+	// Port 0: 8 bit I/O. Shared with d0-d7
+	devcb_read8    m_port0_read;
+	devcb_write8   m_port0_write;
+
+	// Port 1: 8 bit I/O. Shared with d8-d15
+	devcb_read8    m_port1_read;
+	devcb_write8   m_port1_write;
+
+	// Port 2: 8 bit I/O. Shared with d16-d23
+	devcb_read8    m_port2_read;
+	devcb_write8   m_port2_write;
```

Comment:
Can you use a `devcb_read8::array` and `devcb_write8::array` for these to make it more concise?